api/v1alpha1/localbuild_types.go

```diff
@@ -45,8 +45,19 @@ type PackageConfigsSpec struct {
 	CorePackageCustomization map[string]PackageCustomization `json:"packageCustomization,omitempty"`
 }
 
+// BuildCustomizationSpec fields cannot change once a cluster is created
+type BuildCustomizationSpec struct {
+	Protocol       string `json:"protocol,omitempty"`
+	Host           string `json:"host,omitempty"`
+	IngressHost    string `json:"ingressHost,omitempty"`
+	Port           string `json:"port,omitempty"`
```

Comment:
Why are we choosing string rather then an int?

Reply:
I wanted to follow the standard package conventions. 

https://pkg.go.dev/net/url#URL.Port

Reply:
Ahh I didn't realize Go put ports as strings...

Reply:
Oh I know... It was a surprise for me too.